### PR TITLE
Add Java 17 to build/test matrix

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

**Comment by basil** [comment]
Used my admin powers to replay the build with the `Jenkinsfile` changes here: https://ci.jenkins.io/job/Plugins/job/ssh-slaves-plugin/job/PR-402/2/

File: Jenkinsfile [hunk 1]
```diff
@@ -1,8 +1,4 @@
-//def buildConfiguration = buildPlugin.recommendedConfigurations()
-
-def buildConfiguration = [
-  [platform: 'windows', jdk: '11'],
-  [platform: 'linux',   jdk: '11'],
-]
-
-buildPlugin(configurations: buildConfiguration, timeout: 90)
+buildPlugin(useContainerAgent: true, configurations: [
+  [platform: 'linux', jdk: 17],
```
Comment (kuisathaverat) [inline Jenkinsfile:2]:
```suggestion
  [platform: 'linux', jdk: 17],
  [platform: 'linux', jdk: 11],
```

Comment (kuisathaverat) [reply Jenkinsfile:2]: I'd preferred to not remove Linux Java 11 build and test

Comment (basil) [reply Jenkinsfile:2]: Inconsistent with the archetype, but OK

**Review by kuisathaverat — APPROVED** [review]

**Review by newtadeocampos-tech — COMMENTED** [review]
T

**Pull request merged**